python/ccf/ledger.py

```diff
@@ -803,6 +803,43 @@ def get_seqnos(self):
         return self.start_seqno, self.end_seqno
 
 
+class LedgerIterator:
+    _filenames: list
+    _fileindex: int
+    _current_chunk: LedgerChunk
+    _ledger_validator: Optional[LedgerValidator] = None
+
+    def __init__(
+        self, filenames: list, ledger_validator: Optional[LedgerValidator] = None
+    ):
+        self._filenames = filenames
+        self._fileindex = -1
+        self._ledger_validator = ledger_validator
+
+    def __next__(self) -> LedgerChunk:
+        self._fileindex += 1
+        if len(self._filenames) > self._fileindex:
+            self._current_chunk = LedgerChunk(
+                self._filenames[self._fileindex], self._ledger_validator
+            )
+            return self._current_chunk
+        else:
+            raise StopIteration
+
+    def signature_count(self) -> int:
```

Comment:
This isn't great, it's not possible to distinguish a ledger that legitimately has no signatures, versus one that does but the validator wasn't set. I think we should remove this call and force the caller to deal with by exposing the validator, or raise an exception.

Reply:
Or possibly make it optional like last_verified_txid below, but avoid overlap between "no value" and 0.